mmv1/third_party/terraform/resources/resource_dns_record_set.go

```diff
@@ -126,36 +127,70 @@ func resourceDnsRecordSet() *schema.Resource {
 											Type: schema.TypeString,
 										},
 									},
+									"health_checked_targets": {
+										Type:        schema.TypeList,
+										Optional:    true,
+										Description: "The list of targets to be health checked. Note that if DNSSEC is enabled for this zone, only one of `rrdatas` or `health_checked_targets` can be set.",
+										MaxItems:    1,
+										Elem:        healthCheckedTargetSchema,
+									},
 								},
 							},
-							ExactlyOneOf: []string{"routing_policy.0.wrr", "routing_policy.0.geo"},
+							ExactlyOneOf:  []string{"routing_policy.0.wrr", "routing_policy.0.geo", "routing_policy.0.primary_backup"},
+							ConflictsWith: []string{"routing_policy.0.enable_geo_fencing"},
 						},
 						"geo": {
+							Type:         schema.TypeList,
+							Optional:     true,
+							Description:  `The configuration for Geo location based routing policy.`,
+							Elem:         geoPolicySchema,
+							ExactlyOneOf: []string{"routing_policy.0.wrr", "routing_policy.0.geo", "routing_policy.0.primary_backup"},
+						},
+						"enable_geo_fencing": {
+							Type:          schema.TypeBool,
+							Optional:      true,
+							Description:   "Specifies whether to enable fencing for geo queries.",
+							ConflictsWith: []string{"routing_policy.0.wrr", "routing_policy.0.primary_backup"},
+						},
+						"primary_backup": {
 							Type:        schema.TypeList,
 							Optional:    true,
-							Description: `The configuration for Geo location based routing policy.`,
+							Description: "The configuration for a primary-backup policy with global to regional failover. Queries are responded to with the global primary targets, but if none of the primary targets are healthy, then we fallback to a regional failover policy.",
+							MaxItems:    1,
 							Elem: &schema.Resource{
 								Schema: map[string]*schema.Schema{
-									"location": {
-										Type:        schema.TypeString,
+									"primary": {
+										Type:        schema.TypeList,
 										Required:    true,
-										Description: `The location name defined in Google Cloud.`,
+										Description: "The list of global primary targets to be health checked.",
+										MaxItems:    1,
+										Elem:        healthCheckedTargetSchema,
 									},
-									"rrdatas": {
-										Type:     schema.TypeList,
-										Required: true,
-										Elem: &schema.Schema{
-											Type: schema.TypeString,
-										},
+									"backup_geo": {
+										Type:        schema.TypeList,
+										Required:    true,
+										Description: "The backup geo targets, which provide a regional failover policy for the otherwise global primary targets.",
+										Elem:        geoPolicySchema,
+									},
+									"enable_geo_fencing_for_backups": {
+										Type:        schema.TypeBool,
+										Optional:    true,
+										Description: "Specifies whether to enable fencing for backup geo queries.",
+									},
+									"trickle_ratio": {
```

Comment:
Note that [gcloud](https://cloud.google.com/sdk/gcloud/reference/dns/record-sets/create) uses `trickle_ratio` and `enabled_geo_fencing` field names as well, even though they do not exactly match the API.